CONTRIBUTING.md

```diff
@@ -338,8 +338,10 @@ This project follows the [Kubernetes API conventions](https://github.com/kuberne
 
 Optional fields have the following properties:
 * An optional field MUST be marked with `+optional` and include an `omitempty` JSON tag.
-* Fields SHOULD be pointers if the nil and the zero values (by Go standards) have semantic differences.
-  * Note: This doesn't apply to map or slice types as they are assignable to `nil`.
+* Fields SHOULD be pointers if there is a good reason for it, e.g:
+  * the nil and the zero values (by Go standards) have semantic differences.
+    * Note: This doesn't apply to map or slice types as they are assignable to `nil`.
+  * the field is of a struct type and would show up as an empty object after marshalling (e.g. `kubectl get`) 
```

Comment:
What about
```suggestion
  * the field is of a struct type containing all omitempty values, and you want to prevent it to show up as an empty object after marshalling (e.g. `kubectl get`) 
```

Reply:
So the empty object wouldn't show up if one of the fields doesn't have `omitempty`? (as that field would then be always rendered and thus the object wouldn't be empty).

I'm not 100% sure if we should couple the decision to the current state of the fields as the fields can change over time but changing something to a pointer is a breaking change. (which I guess requires deprecating the whole field)

Reply:
This kinda comes down to generic kube recommendation, if it's optional make it a pointer.

Reply:
An optional struct might have required fields but still be a pointer though?

Reply:
> This kinda comes down to generic kube recommendation, if it's optional make it a pointer.

I would only consider this for structs though, for basic types it's imho too painful to handle and has no upside if you don't care about the difference between unset and the default value.

> An optional struct might have required fields but still be a pointer though?

Yup exactly. I think that would be fine too.

Reply:
Btw, it's a nuance, but I intentionally wrote `e.g.`. So the general idea is, hey please let's only make these a pointer if there is a reason for it (aka we know what we're doing). It's not meant as an exhaustive list.

Reply:
Thanks Stefan, let's incorporate Fabrizio suggestion and proceed.

Reply:
Rephrased accordingly. @fabriziopandini @enxebre @vincepri PTAL :)